src/System.Collections.NonGeneric/src/System/Collections/Hashtable.cs

```diff
@@ -141,6 +141,79 @@ private struct bucket
         private IEqualityComparer _keycomparer;
         private Object _syncRoot;
 
+        [Obsolete("Please use EqualityComparer property.")]
+        protected IHashCodeProvider hcp
+        {
+            get
+            {
+                if (_keycomparer is CompatibleComparer)
+                {
+                    return ((CompatibleComparer)_keycomparer).HashCodeProvider;
+                }
+                else if (_keycomparer == null)
+                {
+                    return null;
+                }
+                else
+                {
+                    throw new ArgumentException(SR.Arg_CannotMixComparisonInfrastructure);
```

Comment:
Yup.  Thanks for keeping me honest :)  I've added tests to cover all this stuff.